src/BatteryStateOfChargeService.h

```diff
@@ -19,4 +27,27 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     const double initialStateOfChargePercent_;
 
     // Recommend adding necessary variables here
+    double initialCurrent;
+    double newCurrent = 0;
+    QTime initialTime;
+    QTime newTime;
+
+    double dTime;
+    double aveCurrent;
+    double dAmpHours;
+    double initialampHours;
+    double timeRemaining;
+    double ampsRemaining;
+    double rateofDepletion;
+    double hoursPrime;
+    double minutesPrime;
+    double secondsPrime;
+    double aveCurrentPrime;
+    double currentPrime = 0;
+    QTime t;
+
+    int count;
```

Comment:
So a lot of these variables are only used in the `void addData(const BatteryData& batteryData)` function.
Putting them as private members in the class implies that the variables will be used in a few functions, that the variable is an important part of the object itself. To fix this, I would suggest remove everything here that is only used in the `addData` function and initialize them in the `addData` function.